### PR TITLE
[experiment] css flexiibility

### PR DESCRIPTION
**Pull request by EdwardIrby**
### Generated dist output
- A postcss plugin generates a css file that then has the fallback values to support ie11
### Outstanding questions
- should this css approach happen before esm?
  - right now for this to work we need to generate the css file in the same relative location in the dist directory.
- do we want to add stylelint ci step to the project?

## Steps
1.  Refactor imported core vars

```
import {
  layout,
  type,
  motion,
  colorsTextIcon,
  colorsGreen,
  colorsBackgroundDark,
  colorsBorder,
  colorsBackgroundUtility
} from '@pluralsight/ps-design-system-core'
```

```
import { refactorCssVars} from '@pluralsight/ps-design-system-core'
const {
  layout,
  type,
  motion,
  colorsTextIcon,
  colorsGreen,
  colorsBackgroundDark,
  colorsBorder,
  colorsBackgroundUtility
} = refactorCssVars
```
2: Run build in package
3. Copy generated css file in dist to `src/css`
4.  Refactor `.storybook/main.js`
```js
module.exports = {
  addons: ['@pluralsight/ps-design-system-storybook-preset'],
  stories: ['../src/**/*.story.@(js|ts|tsx)'],
  typescript: {
    check: true,
    reactDocgen: 'react-docgen-typescript'
  }
}
```

```js
const path = require('path');
const styleLoader = require.resolve('style-loader')
const cssLoader = require.resolve('css-loader')
const postcssLoader = require.resolve('postcss-loader')

module.exports = {
  addons: ['@pluralsight/ps-design-system-storybook-preset'],
  stories: ['../src/**/*.story.@(js|ts|tsx)'],
  typescript: {
    check: true,
    reactDocgen: 'react-docgen-typescript'
  },
  webpackFinal: (config) => {
    config.module.rules = config.module.rules.filter(
      f => f.test.toString() !== '/\\.css$/'
    );
    config.module.rules.push({
      test: /\.css$/,
      sideEffects: true,
      use: [styleLoader, {
        loader: cssLoader,
        options: {
          import: true,
        }
      }, {
        loader: postcssLoader,
        options: {
          config: {
            path: path.resolve(__dirname, '../'),
          }
        },
      }],
      include: [
        path.resolve(__dirname, '../'),
        path.resolve(__dirname, '../../normalize/dist/index.css')
      ],
    });
    return config;
  },
}
```
5. add postcss.config.js
```js
module.exports = {
  plugins: [
    require('postcss-nesting'),
    require('postcss-preset-env')({
      importFrom: [
        '../core/dist/css/breakpoints.module.css',
        '../core/dist/css/colors.module.css',
        '../core/dist/css/layers.module.css',
        '../core/dist/css/layout.module.css',
        '../core/dist/css/type.module.css',
        '../core/dist/css/motion.module.css'
      ]
    })
  ]
}
```
6. add build-css.js
```js
const { execSync } = require('child_process')
const { resolve, join } = require('path')
const { promises: fs } = require('fs')

const root = resolve(__dirname)
const run = cmd => execSync(cmd, { stdio: 'inherit', cwd: root })
const build = async () => {
  try {
    await fs.mkdir(join(__dirname, './dist/cjs/css/'), { recursive: true })
    console.log('Directory created successfully!')
  } catch (err) {
    return console.error(err)
  }

  run('postcss src/css/index.css -o dist/esm/css/index.css')
  run('cp dist/esm/css/index.css dist/cjs/css/index.css ')
  run(
    'postcss src/css/index.css -o dist/css/index.css && cp dist/css/index.css dist/css/index.scss'
  )
}
build()

```
7. update `build:css`
```json
"build:css": "node ./build-css.js"
```
8. Swap out glamor `css` and `compose` for classNames util in react components 
9. Refactor spreads to to use className 
```js
<div {...styles.class(props)} />
```

```js
<div className={styles.class(props)} />
```

10.  Refactor root node like so
  ```js
<div {...styles.class(props)} />
```

```js
<div className={classNames(styles.class(props), className)} />
```
10. Clean up CSS file now that we don't use glamor

---

Once all packages have been refactored. Refactor build script to use build-css.js code.

### CODE REVIEW

File: packages/link/postcss.config.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    require('autoprefixer'),
+    require('postcss-nesting'),
+    require('postcss-custom-properties')({
+      importFrom: [
+        '../core/dist/css/breakpoints.module.css',
+        '../core/dist/css/colors.module.css',
+        '../core/dist/css/layers.module.css',
+        '../core/dist/css/layout.module.css',
+        '../core/dist/css/type.module.css',
+        '../core/dist/css/motion.module.css'
```
Comment (EdwardIrby) [inline packages/link/postcss.config.js:12]: This is necessary for correct postcss build normalize doesn't solve this if we want ie11 fallback.

**Review by sweeting — COMMENTED** [review]
I can't really speak to the technical stuff, but those `psds-` class names seem much easier to target for css overrides.

**Comment by jaketrent** [comment]
I think we can close this POC at this point. What do you think?

**Comment by jaketrent** [comment]
This is currently unused. Closing.

**Pull request closed without merging**